docs/platform-engineers/cue/advanced.md

```diff
@@ -2,5 +2,453 @@
+
+	$ vela components
+	NAME       	NAMESPACE  	WORKLOAD                             	DESCRIPTION
+	...
+	my-stateful	vela-system	statefulsets.apps                    	My StatefulSet component.
+	... 
+
+By publishing the `Application` of KubeVela to the cluster, we can pull up the StatefulSet and Service objects we just defined.
+
+	cat <<EOF | kubectl apply -f -
+	apiVersion: core.oam.dev/v1beta1
+	kind: Application
+	metadata:
+	  name: website
+	spec:
+	  components:
+	    - name: my-component
+	      type: my-stateful
+	EOF
+
+
+
+## Define Customized Parameters For Component
+
+In order to meet the changing needs of users, we need to expose some parameters in the last `parameter`. You can learn about the syntax of parameters in [CUE Basic][7].
+
+In this example, we expose the following parameters to the user:
+
+* Image name, allowing users to customize the image
+* Instance name, allowing users to customize the instance name of the generated StatefulSet object and Service object
+* The number of copies, the number of copies of the generated object
+
+
+		... # Omit other unmodified fields
+		template: {
+			output: {
+				apiVersion: "apps/v1"
+				kind:       "StatefulSet"
+				metadata: name: parameter.name
+				spec: {
+					selector: matchLabels: app: "nginx"
+					replicas:    parameter.replicas
+					serviceName: "nginx"
+					template: {
+						metadata: labels: app: "nginx"
+						spec: {
+							containers: [{
+								image: parameter.image
+			
+							    ... // Omit other unmodified fields
+							}]
+						}
+					}
+					    ... // Omit other unmodified fields
+				}
+			}
+			outputs: web: {
+				apiVersion: "v1"
+				kind:       "Service"
+				metadata: {
+					name: "nginx"
+					labels: app: "nginx"
+				}
+				spec: {
+					... // Omit other unmodified fields	
+			    }
+			}
+			parameter: {
+				image: string
+				name: string
+				replicas: int
+			}
+		}
+
+After modification, use `vela def apply` to install to the cluster:
+
+	$ vela def apply my-stateful.cue
+	ComponentDefinition my-stateful in namespace vela-system updated.
+
+This modification process takes effect in real time, and the user can immediately see that the my-stateful component in the system has added new parameters.
+
+	$ vela show my-stateful
+	# Properties
+	+----------+-------------+--------+----------+---------+
+	|   NAME   | DESCRIPTION |  TYPE  | REQUIRED | DEFAULT |
+	+----------+-------------+--------+----------+---------+
+	| name     |             | string | true     |         |
+	| replicas |             | int    | true     |         |
+	| image    |             | string | true     |         |
+	+----------+-------------+--------+----------+---------+
+
+The modification of the ComponentDefinition will not affect the application that is already running. The new ComponentDefinition will take effect when the application is modified and redeployed next time.
```

Comment:
```suggestion
Updating the ComponentDefinition will not affect existing Applications. It will take effect only after updating the Applications next time.
```